app/models/patient.rb

```diff
@@ -294,4 +302,9 @@ def self.fulfilled_on_or_before(datetime)
     Patient.where('fulfillment.fulfilled' => true,
                   updated_at: { '$lte' => datetime })
   end
+
+  def calculate_date_to_be_archived
```

Comment:
Since this is setting `archive_date` to initial call date plus a year, should we just make it a method a la `obsolete?` rather than saving it as a db field and setting that with a callback?  e.g.:

```
def archive_date
  initial_call_date + 1.year
end
```